homeassistant/components/dlna_dmr/config_flow.py

```diff
@@ -130,32 +135,56 @@ async def async_step_ssdp(self, discovery_info: ssdp.SsdpServiceInfo) -> FlowRes
             return self.async_abort(reason="alternative_integration")
 
         # Abort if the device doesn't support all services required for a DmrDevice.
-        # Use the discovery_info instead of DmrDevice.is_profile_device to avoid
-        # contacting the device again.
-        discovery_service_list = discovery_info.upnp.get(ssdp.ATTR_UPNP_SERVICE_LIST)
-        if not discovery_service_list:
+        if not _is_dmr_device(discovery_info):
             return self.async_abort(reason="not_dmr")
 
-        services = discovery_service_list.get("service")
-        if not services:
-            discovery_service_ids: set[str] = set()
-        elif isinstance(services, list):
-            discovery_service_ids = {service.get("serviceId") for service in services}
-        else:
-            # Only one service defined (etree_to_dict failed to make a list)
-            discovery_service_ids = {services.get("serviceId")}
-
-        if not DmrDevice.SERVICE_IDS.issubset(discovery_service_ids):
-            return self.async_abort(reason="not_dmr")
-
-        # Abort if another config entry has the same location, in case the
-        # device doesn't have a static and unique UDN (breaking the UPnP spec).
-        self._async_abort_entries_match({CONF_URL: self._location})
+        # Abort if another config entry has the same location or MAC address, in
+        # case the device doesn't have a static and unique UDN (breaking the
+        # UPnP spec).
+        for entry in self._async_current_entries(include_ignore=True):
+            if self._location == entry.data[CONF_URL]:
+                return self.async_abort(reason="already_configured")
+            if self._mac and self._mac == entry.data.get(CONF_MAC):
+                return self.async_abort(reason="already_configured")
 
         self.context["title_placeholders"] = {"name": self._name}
 
         return await self.async_step_confirm()
 
+    async def async_step_ignore(self, user_input: Mapping[str, Any]) -> FlowResult:
+        """Ignore this config flow, and add MAC address as secondary identifier.
+
+        Not all DMR devices correctly implement the spec, so their UDN may
+        change between boots. Use the MAC address as a secondary identifier so
```

Comment:
Why should we use this as a secondary identifier. If we're going to get the MAC, why wouldn't we always use it as identifier ?

Reply:
As you identified below, the MAC address is not reported by the DLNA API, and the [developer docs](https://developers.home-assistant.io/docs/config_entries_config_flow_handler#unique-id-requirements) state good reasons not to use it as a primary unique ID in this case. 

Unfortunately some devices purport to be DLNA compliant but break from the spec in some ways, and this PR is adding a heuristic to deal with certain cases of breakage.